server/src/main/java/org/opensearch/action/admin/cluster/health/TransportClusterHealthAction.java

```diff
@@ -283,6 +283,7 @@ private void executeHealth(
             if (request.ensureNodeWeighedIn()) {
                 if (clusterHealthResponse.hasDiscoveredClusterManager() == false) {
                     listener.onFailure(new ClusterManagerNotDiscoveredException("cluster-manager not discovered"));
+                    return;
```

Comment:
Does this mean that the previous code would have invoked both `onFailure` and `onResponse` on the listener?

Reply:
No the listener would have unblocked the original request but would have continued executing requests, invoking other listeners on the way on a closed connection. Benign but wasteful

Reply:
Understood. I'm just wondering if there would be any value to create a base listener implementation that would fail fast if more than one callback is invoked. What do you think?

Reply:
+1 lets open an issue?